Client.go

```diff
@@ -21,28 +123,36 @@ type ClientConfig struct {
 	Passwd   string
 	Token    string
 }
+*/
 
 // Client hold rest api configuration and http connection
 type Client struct {
-	cfg        ClientConfig
+	cfg        *spvConfig
 	HTTPClient HTTPClient
 }
 
 // NewClient create a new rest Client
-func NewClient(c ClientConfig) *Client {
+func NewClient(opts ...SPVConfigFunc) *Client {
```

Comment:
Improve the docs here, add a list of the required settings and list them out - try to be verbose here, this will be used by the public so good to over do documentation than not.